### PR TITLE
feature: Add Donation Summary to form.

### PR DESCRIPTION
**Pull request by kjohnson**
## Description

This PR adds a "donation summary" section to the form, which includes the following:

- Donation amount
- Donation frequency
- Donation fees (if applicable)
- Donation total

## Demo

A demo sandbox is available for ease of testing: https://light-caiman.w5.wpsandbox.pro/

## Known Issues

- [x] Sometimes, switching the payment gateway removes the donation summary... 🤦🏻
#6064
- [x] Responsive styles
#6095 

### Template Location
| Before Payment Fields | After Payment Fields |
| --- | --- |
| image | image |

### Form Settings

| Enabled for new forms | Disabled for existing forms |
| --- | --- |
| image | image |

### GiveWP

image

### Fee Recovery Add-on

image

### Reucrring Donations Add-on

| Opt-out | Opt-in |
| --- | --- |
| image | image |

#### Admin Defined and Variable Recurring Duration

| | |
| --- | --- |
| Screen Shot 2021-11-04 at 14 59 28 | Screen Shot 2021-11-04 at 14 59 56 |

### Currency Switcher Add-on

| Pound Sterling | Japanese Yen |
| --- | --- |
| image | image  |

### Responsive Styling

| | |
| --- | --- |
| image | image |
| image | image |

## Testing Instructions

- Supports Recurring by updating the donation frequency
- Supports Fee Recovery by displaying the fee amount and the total amount
- Supports Currency Switcher by updating the currency in the summary (amount, fee, and total)
- Supports Currency Switcher with zero decimal currencies (ie Japanese Yen)
- Supports Currency Switcher with switched separators (ie Euros)

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Review by DevinWalker — CHANGES_REQUESTED** [review]
Hey @kjohnson I just did a review of this PR using the sandbox you created and here's my feedback: https://www.loom.com/share/13f9e1e6c2494c14879e86eb26791e62

Please let me know if you have any questions and thanks for the hard work!

**Comment by kjohnson** [comment]
- [ ] Hyperlinks should use form primary color
- [ ] Capitalize "edit donation"
- [ ] Add pencil icon to "edit donation"
- [ ] Co-locate settings with related step
- [ ] Add semi-colon to description default value
- [ ] Add "Leave blank to remove" for description
- [ ] Stack elements on smaller screens
- [ ] Recurring hyperlink font size
- [ ] Support admin defined and per tier periods

File: src/DonationSummary/resources/views/summary.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    <div class="give-donation-summary-table-wrapper">
+
+        <table>
+            <thead>
+                <tr>
+                    <th>Donation Summary</th>
+                    <th>
+                        <button class="back-btn" onclick="GiveDonationSummary.handleNavigateBack(event)">
+                            Edit Donation<?php include plugin_dir_path( __DIR__ ) . 'images/pencil.svg'; ?>
```
Comment (ravinderk) [inline src/DonationSummary/resources/views/summary.php:20]: @kjohnson can we make text translatable in this view?

**Review by DevinWalker — CHANGES_REQUESTED** [review]
Hey @kjohnson I just reviewed it again and it's looking great! The one thing I saw was the way the summary looks on smaller screens isn't what I was expecting. I was thinking the line items would stack and be centered. Let me know if you agree or if you were thinking something else with your implementation.

Here's my testing video: https://www.loom.com/share/0db23939b6244cada4ed14da3f713047

**Comment by kjohnson** [comment]
@DevinWalker the heading wasn't showing because I had moved (changed the name of) the setting, so you would have had to re-publish the get it to apply - so that won't be an issue for new donation forms.

**Comment by kjohnson** [comment]
@DevinWalker the mobile styles that I added respected the original text-alignment in the description list format. What you are seeing are the rows shifting between each other, like a zipper. That said, I can update it to be centered if you would prefer.

**Comment by kjohnson** [comment]
@DevinWalker I have updated the responsive styles from "stacked" to "compact"

| | |
| --- | --- |
| image | image |
| image | image |

**Review by DevinWalker — APPROVED** [review]
@kjohnson this looks great now 👍

**Comment by JasonTheAdams** [comment]
@kjohnson I just started testing this. I pulled down the PR and checked an existing multi-step form. It _is_ showing the Donation Summary — shouldn't it not be for existing forms?

**Comment by kjohnson** [comment]
@JasonTheAdams looks like something got lost in a merge - it should be fixed now, see https://github.com/impress-org/givewp/pull/6061/commits/1efb0a4e08fdba98d723eaf49f67c249e3aaf95b .

**Comment by JasonTheAdams** [comment]
Thanks, @kjohnson! It looks like it's working, but I'm getting some kind of console error:
image

**Review by JasonTheAdams — APPROVED** [review]
Just excellent work, @kjohnson! The UI looks great and is nicely responsive, and the code is well organized and well written. 👍 

I just have the one comment above for an error in the console, though it doesn't seem to be breaking anything. Please clean that up and merge this up!

**Comment by kjohnson** [comment]
@JasonTheAdams I'm seeing a slightly different console error, which I have resolved. Can you confirm that this is the same error? The line numbers are similar.

**Review by JasonTheAdams — APPROVED** [review]
Working great! I'm excited to get this out! 🎉

**Pull request merged**